modules/geolocation/geolocation.py

```diff
@@ -96,6 +95,42 @@ def gather_point_pairs(self):
             pixelGeoPairs = np.concatenate((pixelGeoPairs, [pair]))
 
         return pixelGeoPairs
+    
+    def get_non_collinear_points(self, coordinatesArray):
+        """
+        Returns a list of four coordinates from an input array that are not collinear to one another.
+
+        Parameters
+        ----------
+        coordinatesArray : np.ndarray
+            Array with dimensions ( , 2), containing a list of coordinates
+
+        Returns
+        -------
+        np.ndarray
+            Array with dimensions (4, 2), containing a list of coordinates that are non-collinear,
+            or an empty list if none were found in the input array
+        """
+        collinearPoints = np.zeros(shape=(4, 2))
+
+        if len(coordinatesArray) < 4:
+            return collinearPoints
+
+        for i in range(0, len(coordinatesArray)-3):
+            x1, y1 = coordinatesArray[i]
+            x2, y2 = coordinatesArray[i+1]
+            x3, y3 = coordinatesArray[i+2]
+            x4, y4 = coordinatesArray[i+3]
+
```

Comment:
I think the readability and brevity of this code could be improved. For starters, make x and y numpy arrays. Then structure lines 120-123 as:
`
for j in range(0, 4):
x[j], y[j] = coordinatesArray[i + j]
`
which should result in the same behaviour as:
x[0], y[0] = coordinatesArray[i]
...
x[3], y[3] = coordinatesArray[i+3]